### DISCUSSION TITLE
Is there an API that can refresh a specific atom?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [pmndrs/jotai#504] [General] [answered]
Is there an API that can refresh a specific atom, something similar to react-query `queryClient.invalidateQueries('todos')`?

**Comment by dai-shi** [comment]
Hi, if it's a sync atom, refreshing means reset, like setting the default value, right?
Or, is it about async atom? If so, more use case description would be helpful.

**Reply by ivanjeremic** [reply]
No not reset, by the way, the concept of resettable, [https://docs.pmnd.rs/jotai/guides/resettable](https://docs.pmnd.rs/jotai/guides/resettable), why is this needed when every atom comes anyway with a setter which you can use to reset the state?

What I really meant is async data when you fetch something from an API.

**Reply by dai-shi** [reply, marked as answer]
> you can use to reset the state?

It's just handy. Not necessary.

> What I really meant is async data when you fetch something from an API.

Okay, async or not, the derived atom can't be reset.
In such use cases, I'd create a new atom, or we can now store a promise as an atom value.
Hm, if you just need to refetch, you could add another dependency.
```js
const version = atom(0)
const data = atom(async (get) => {
  get(version)
  const response = await fetch(...)
  return response.json()
})
```
